examples/cmd/snapshotting/example_demo.go

```diff
@@ -0,0 +1,410 @@
+	}
+	defer func() {
+		if err := m.StopVMM(); err != nil {
+			log.Fatal(err)
+		}
+	}()
+	defer func() {
+		if err := m.Shutdown(ctx); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	ipToRestore = m.Cfg.NetworkInterfaces[0].StaticConfiguration.IPConfiguration.IPAddr.IP.String()
+	fmt.Printf("IP of VM: %v\n", ipToRestore)
+
+	sshKeyPath := filepath.Join(dir, "root-drive-ssh-key")
+
+	var client *ssh.Client
+	for i := 0; i < maxRetries; i++ {
+		client, err = connectToVM(m, sshKeyPath)
+		if err != nil {
+			time.Sleep(backoffTimeMs * time.Millisecond)
+		} else {
+			break
+		}
+	}
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer client.Close()
+
+	session, err := client.NewSession()
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer session.Close()
+
+	fmt.Println(`Sending "sleep 422" command...`)
+	err = session.Start(`sleep 422`)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println("Creating snapshot...")
+
+	err = m.PauseVM(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = m.CreateSnapshot(ctx, memPath, snapPath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = m.ResumeVM(ctx)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println("Snapshot created")
+}
+
+func loadSnapshotSSH(ctx context.Context, socketPath, memPath, snapPath string) {
+	var ipFreed bool = false
+	var err error
+
+	for i := 0; i < maxRetries; i++ {
+		// Wait till the file no longer exists (i.e. os.Stat returns an error)
+		if _, err = os.Stat(fmt.Sprintf("%s/networks/%s/%s", cniCacheDir, networkName, ipToRestore)); err == nil {
+			time.Sleep(backoffTimeMs * time.Millisecond)
+		} else {
+			ipFreed = true
+			break
+		}
+	}
+
+	if errors.Is(err, os.ErrNotExist) {
+		err = nil
+	} else if !ipFreed {
+		err = fmt.Errorf("IP %v was not freed", ipToRestore)
+	}
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	dir, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	cniConfDir := filepath.Join(dir, "cni.conf")
+	cniBinPath := []string{filepath.Join(dir, "bin")} // CNI binaries
+
+	// Network config, using the previous machine's IP
+	cniConfPath := fmt.Sprintf("%s/%s.conflist", cniConfDir, networkName)
+	err = writeCNIConfWithHostLocalSubnet(cniConfPath, networkName, subnet)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.Remove(cniConfPath)
+
+	networkInterface := sdk.NetworkInterface{
+		CNIConfiguration: &sdk.CNIConfiguration{
+			NetworkName: networkName,
+			IfName:      ifName,
+			ConfDir:     cniConfDir,
+			BinPath:     cniBinPath,
+			Args:        [][2]string{{"IP", ipToRestore + networkMask}},
+			VMIfName:    "eth0",
+		},
+	}
+
+	cfg := createNewConfig(fmt.Sprintf("%s.load", socketPath),
+		withNetworkInterface(networkInterface),
+	)
+
+	// Use the firecracker binary
+	cmd := sdk.VMCommandBuilder{}.WithSocketPath(fmt.Sprintf("%s.load", socketPath)).WithBin(filepath.Join(dir, "firecracker")).Build(ctx)
+
+	m, err := sdk.NewMachine(ctx, cfg, sdk.WithProcessRunner(cmd))
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.Remove(socketPath + ".load")
```

Comment:
nit: can we have a variable `... = fmt.Sprintf("%s.load", socketPath)` and use that 3x ? similar comment for the create snapshot function.